### PR TITLE
Add support for Eastern Arabic numerals

### PR DESCRIPTION
**Pull request by bluebear94**
See #3990.

Add both the Arabic and Persian variants.

This PR also changes the `zeroless` function to use a `SmallVec` for accumulating the digits.

### CODE REVIEW

File: tests/suite/model/numbering.typ [hunk 1]
```diff
@@ -98,6 +98,22 @@
   [ for #i \ ]
 }
 
+--- numbering-arabic-indic ---
```
Comment (laurmaedje) [inline tests/suite/model/numbering.typ:101]: I forgot to adjust the numbering tests when reworking the test suite, but I don't think these should be image-based anymore. The other tests can be changed in a separate PR, but we can start with assertion-based tests here.

Comment (bluebear94) [reply tests/suite/model/numbering.typ:101]: Okay, I’ve changed this test to use assertions.

**Comment by laurmaedje** [comment]
Thanks!
